### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by sgnoohc**
The main addition is the efficiency code.
The SDL/ codes are largely unchanged.
Most of the changes are at the application side.
See docs/20201222.md for benchmarks.

The other PR open from Bei will likely clash with this in "Makefile" only.

### CODE REVIEW

**Comment by slava77** [comment]
`Files changed 660`
is everything really used/needed?

how long does it take to compile?

At this rate I doubt very much that we can consider proposing this as an external for CMSSW.

File: docs/20201222.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Here I document the GPU and CPU version benchmark for low pt muon gun efficiencies with GPU and CPU.
+
+### Copy-paste command lines
+
+Log on to phi3
+
+    git clone git@github.com:sgnoohc/TrackLooper-1.git
+    cd TrackLooper-1/
+    git checkout 463ec57
```
Comment (slava77) [inline docs/20201222.md:11]: is this current?

I was expecting instructions based on the SegmentLinking/TrackLooper repo

Comment (sgnoohc) [reply docs/20201222.md:11]: It's a bootstrap paradox.
I can't point to the main repo tag unless we finish this pull request.

Comment (slava77) [reply docs/20201222.md:11]: shouldn't it at worst point to this PR `sgnoohc:master`?
... btw, it's not a good idea to make a PR using `master` as a topic branch.

**Comment by sgnoohc** [comment]
Closing this PR as I would like to make some more extensive changes.

**Pull request closed without merging**